rust/generated_parser/src/ast_builder.rs

```diff
@@ -1179,6 +1179,128 @@ impl<'alloc> AstBuilder<'alloc> {
         ))
     }
 
+    // MemberExpression : MemberExpression TemplateLiteral
+    // CallExpression : CallExpression TemplateLiteral
+    pub fn optional_tagged_template_expr(
+        &self,
+        tag: arena::Box<'alloc, Expression<'alloc>>,
+        mut template_literal: arena::Box<'alloc, TemplateExpression<'alloc>>,
+    ) -> arena::Box<'alloc, Expression<'alloc>> {
+        template_literal.tag = Some(tag);
+        self.alloc(Expression::TemplateExpression(template_literal.unbox()))
+    }
+
+    // MemberExpression : MemberExpression TemplateLiteral
+    // CallExpression : CallExpression TemplateLiteral
+    pub fn optional_tagged_template_expr_tail(
+        &self,
+        template_literal: arena::Box<'alloc, TemplateExpression<'alloc>>,
+    ) -> arena::Box<'alloc, Expression<'alloc>> {
+        self.alloc(Expression::TemplateExpression(template_literal.unbox()))
```

Comment:
also, i suspect this is wrong

Reply:
Yes, please change the name of this method to something like `error_optional_chain_with_template`, make the return type a `Result`, and just always return `Err(ParseError::SyntaxError(template_literal))` or IllegalCharacter or whatever. Or add a new error message in generated_parser/src/error.rs (which is not generated, yes we should figure this out soon). We do not have a way to use the error messages in js.msg from Rust yet.